python/paddle/distributed/auto_parallel/static/pir_pass.py

```diff
@@ -1066,20 +1060,48 @@ def complete_chunk_id(dist_program, startup_program, pipeline_strategy):
     # Step3: Get op index boundary, pp_stage, chunk_id, struct_names of each segment
     seg_pp_stages = [i % pp_degree for i in range(num_chunks)]
     seg_chunk_ids = [i // pp_degree for i in range(num_chunks)]
-    seg_layer_num = [0] * num_chunks
-    for j in range(0, len(seg_struct_names)):
-        i = j % num_chunks
-        seg_layer_num[i] = seg_layer_num[i] + 1
     seg_parts = [0]
-
+    last_struct_name = None
+    user_layer_to_stage_id = (
+        []
+    )  # User intent - a list corresponding to the model layer and the device. The key of the list is the number of the layer, and the value is the corresponding pp_stage.
     for idx, op in enumerate(ops):
         if len(seg_parts) == len(seg_struct_names):
             break
         struct_name = _extract_seg_method(op, seg_method)
+        if op.dist_attr is not None and last_struct_name != struct_name:
+            # When traversing the operations, filter out those without any ops where `has_attr` is `None`. At the same time, ensure that the `pp_stage` of each layer is recorded only once according to the `struct_name`.
+            if (
+                get_pp_stage_by_process_mesh(
+                    op.dist_attr.process_mesh, pp_degree
+                )
+                is not None
+            ):
+                user_layer_to_stage_id.append(
+                    get_pp_stage_by_process_mesh(
+                        op.dist_attr.process_mesh, pp_degree
+                    )
+                )
+            last_struct_name = struct_name
         if struct_name == seg_struct_names[len(seg_parts)]:
             seg_parts.append(idx)
     seg_parts.append(len(ops))
-
+    pp_stage_layer_num = [0] * pp_degree
+    for i in user_layer_to_stage_id:
+        pp_stage_layer_num[i] = pp_stage_layer_num[i] + 1
+    assert all(
+        value >= vpp_degree for value in pp_stage_layer_num
+    ), "The number of layers on each pp_stage must not be less than the vpp_degree in the pp_stage to ensure that each chunk contains at least one layer."
+    seg_layer_num = [0] * num_chunks
+    for pp_stage in range(
+        0, pp_degree
+    ):  # Each pp_stage is assigned a number of tiers based on user intent.
+        pp_stage_layer_nums = pp_stage_layer_num[pp_stage]
```

Comment:
```suggestion
        pp_stage_layer_num = pp_stage_layer_nums[pp_stage]
```
变量命名要符合语法逻辑，比如加s表示复数

Reply:
Done